### ISSUE TITLE
Double ridge in fine calibration of Cherenkov channels

### ISSUE THREAD

**Issue by Nikolajal** [sipm4all/sipm4eic-testbeam2023-analysis#9]
During the calibration procedure, for `device` 193 `cindex` 512 a double ridge was found in the see saw histogram.

**Comment by Nikolajal** [comment]
During the search for the culprit, a small bug was found: https://github.com/sipm4all/sipm4eic-testbeam2023-analysis/blob/c203869e01478ffe2cd35d99b348ac1ad34ef482/lib/data.h#L91C12-L91C24

Here `fine_offset()` should also be multiplied by `coarse_to_ns`

**Comment by Nikolajal** [comment]
During the search for the culprit, I found that in `data.h` there is a possible legacy code: https://github.com/sipm4all/sipm4eic-testbeam2023-analysis/blob/c203869e01478ffe2cd35d99b348ac1ad34ef482/lib/data.h#L30C17-L30C38

There is a duplicate of fine tune calibration loading/calculation

**Comment by Nikolajal** [comment]
During the search for the culprit, I found that in `light data.h` there is a possible small bug: https://github.com/sipm4all/sipm4eic-testbeam2023-analysis/blob/c203869e01478ffe2cd35d99b348ac1ad34ef482/lib/lightdata.h#L43C1-L43C10

In the function:

```
float
lightdata::time() const
{
  auto ci = cindex();
  auto di = device - 192;
  float corr = (float)fine * fine_iif[di][ci] + fine_off[di][ci];
  if (fine >= std::round(fine_cut[di][ci])) corr -= 1.;
  auto time = (float)coarse - corr;
  return time;
}
```

If the calibration is  IIF = 0, CUT = 0, OFF = 0 then `corr` will always be -1 (`corr` is zero from the first line, `fine` will always be greater or equal to zero, thus making `corr = -1.`)
As we search for a potential one clock cycle offset this could be a culprit, even though the event of a null calibration is unlikely if everything goes according to plan.

**Comment by preghenella** [comment]
I don't think any of this is responsible of the double peak I observed.
do you also observe it?

**Comment by preghenella** [comment]
> During the search for the culprit, I found that in `light data.h` there is a possible small bug: https://github.com/sipm4all/sipm4eic-testbeam2023-analysis/blob/c203869e01478ffe2cd35d99b348ac1ad34ef482/lib/lightdata.h#L43C1-L43C10
> 
> In the function:
> 
> ```
> float
> lightdata::time() const
> {
>   auto ci = cindex();
>   auto di = device - 192;
>   float corr = (float)fine * fine_iif[di][ci] + fine_off[di][ci];
>   if (fine >= std::round(fine_cut[di][ci])) corr -= 1.;
>   auto time = (float)coarse - corr;
>   return time;
> }
> ```
> 
> If the calibration is IIF = 0, CUT = 0, OFF = 0 then `corr` will always be -1 (`corr` is zero from the first line, `fine` will always be greater or equal to zero, thus making `corr = -1.`) As we search for a potential one clock cycle offset this could be a culprit, even though the event of a null calibration is unlikely if everything goes according to plan.

I agree that this is not the intended behaviour, I will fix it by checking `fine_cut[di][ci] > 0`

**Comment by preghenella** [comment]
> During the search for the culprit, I found that in `data.h` there is a possible legacy code: https://github.com/sipm4all/sipm4eic-testbeam2023-analysis/blob/c203869e01478ffe2cd35d99b348ac1ad34ef482/lib/data.h#L30C17-L30C38
> 
> There is a duplicate of fine tune calibration loading/calculation

that is no duplicate, it was just old code copy-pasted.
it should be updated to comply with the current IIF, CUT, OFF calibration.
I will take care

**Comment by preghenella** [comment]
> During the search for the culprit, a small bug was found: https://github.com/sipm4all/sipm4eic-testbeam2023-analysis/blob/c203869e01478ffe2cd35d99b348ac1ad34ef482/lib/data.h#L91C12-L91C24
> 
> Here `fine_offset()` should also be multiplied by `coarse_to_ns`

as commented above, it depends on what was the old calibration strategy.
indeed it hat to be updated and I will open a PR to fix this issue. 
Thanks
